ttnn/cpp/ttnn/operations/pool/generic/generic_pools.cpp

```diff
@@ -103,22 +102,45 @@ static Tensor pool2d_invoke(
         // tt::log_debug(tt::LogOp, "auto sharding spec: {}", parallel_config.shard_scheme);
         num_cores_nhw = conv::get_num_cores_nhw_from_parallel_config(parallel_config);
         num_cores_c = conv::get_num_cores_channels_from_parallel_config(parallel_config);
+
+        uint32_t input_channels_alignment = is_in_tiled ? tt::constants::TILE_WIDTH : 8;
+        if (!is_in_tiled && input_tensor.memory_config().is_sharded() &&
```

Comment:
on convs we allow input_channels_alignment to be 8 for BS now as well, I think it woudl work here as well

Reply:
here it should work for any sharding I think